src/batou/tests/test_dependencies.py

```diff
@@ -97,32 +100,38 @@ def test_consumer_retrieves_value_from_provider_order2(env):
 
 def test_consumer_without_provider_raises_error(env):
     env.add_root("consumer", Host("host", env))
-    with pytest.raises(Exception):
-        env.configure()
+    assert len(env.configure()) > 0
     for exc in env.exceptions:
         if isinstance(exc, UnsatisfiedResources):
-            assert set(["the-answer"]) == set(exc.resources)
+            assert set(["the-answer"]) == set(
+                [key for key, _ in exc.unsatisfied_resources]
+            )
             break
     else:
         assert False, "Did not find exception"
 
 
 def test_aggressive_consumer_raises_unsatisfiedrequirement(env):
     env.add_root("aggressiveconsumer", Host("host", env))
-    with pytest.raises(Exception):
-        env.configure()
+    assert len(env.configure()) > 0
     for exc in env.exceptions:
         if isinstance(exc, UnsatisfiedResources):
-            assert set(["the-answer"]) == set(exc.resources)
+            assert set(["the-answer"]) == set(
+                [key for key, _ in exc.unsatisfied_resources]
+            )
             break
     else:
         assert False, "Did not find expected exception."
 
 
 def test_broken_component_logs_real_exception(env):
-    env.add_root("broken", "host")
-    with pytest.raises(UnknownComponentConfigurationError):
-        env.configure()
+    env.add_root("broken", Host("host", env))
+    errors = env.configure()
+    assert len(errors) == 2
+    # could be either one of the two exceptions
+    error_types = (UnknownComponentConfigurationError, NonConvergingWorkingSet)
```

Comment:
This is surprising. I wonder whether NonConvergingWorkingSet should not be reported - in this case. I'm not sure whether they can happen without any other (Non-SilentConfigurationErrors) happening and thus we might want to only report them if they appear alone. Let's make a follow-up ticket for this.

Reply:
#325